catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/TransformersCommand.java

```diff
@@ -52,95 +55,186 @@
 
   private static final String NOT_AVAILABLE = "N/A";
 
+  @Option(
+    name = "--input",
+    required = false,
+    aliases = {"-i", "-input"},
+    multiValued = false,
+    description = "Displays only input transformers"
+  )
+  private boolean inputOption = false;
+
+  @Option(
+    name = "--metacard",
+    required = false,
+    aliases = {"-m", "-metacard"},
+    multiValued = false,
+    description = "Displays only metacard transformers"
+  )
+  private boolean metacardOption = false;
+
+  @Option(
+    name = "--all",
+    required = false,
+    aliases = {"-a", "-all"},
+    multiValued = false,
+    description = "Displays all the properties of the desired transformers"
+  )
+  private boolean allOption = false;
+
   @Override
   protected Object executeWithSubject() throws Exception {
 
-    // ServiceReferences are needed to get property information from InputTransformers
-    List<ServiceReference<InputTransformer>> serviceReferences =
-        (List<ServiceReference<InputTransformer>>)
-            bundleContext.getServiceReferences(InputTransformer.class, "(id=*)");
+    if (inputOption) {
+      printTransformers(InputTransformer.class, INPUT);
+      return null;
+    }
+    if (metacardOption) {
+      printTransformers(MetacardTransformer.class, METACARD);
+      return null;
+    }
+
+    printTransformers(InputTransformer.class, INPUT);
+    printTransformers(MetacardTransformer.class, METACARD);
+
+    return null;
+  }
+
+  private void printTransformers(Class transformerClass, String type) throws Exception {
+
+    Collection<ServiceReference> sref =
+        bundleContext.getServiceReferences(transformerClass, "(id=*)");
 
     List<TransformerProperties> transformersProperties =
-        serviceReferences.stream().map(TransformerProperties::new).collect(Collectors.toList());
+        sref.stream().map(TransformerProperties::new).collect(Collectors.toList());
 
     int activeTransformers = transformersProperties.size();
 
     if (activeTransformers == 0) {
```

Comment:
❓ Couldn't you have changed NO_ACTIVE_TRANSFORMERS to "There are no active %s transformers" and just called String.format(NO_ACTIVE_TRANSFORMERS, type) instead of getNoActiveString(type)?